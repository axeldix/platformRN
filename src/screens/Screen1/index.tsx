import {type FC, memo} from 'react';

import {useNavigation} from '../../utils';
import {Wrapper, Title, StyledButton, StyledButtonText} from './styled';

const Screen1: FC = () => {
  const {navigate} = useNavigation();

  const screenName = 'Screen1';

  return (
    <Wrapper>
      <Title>{screenName}</Title>
      <StyledButton onPress={() => navigate('Screen2')}>
        <StyledButtonText>{'RN Navigation -> Screen 2'}</StyledButtonText>
      </StyledButton>
    </Wrapper>
  );
};

export default memo(Screen1);
