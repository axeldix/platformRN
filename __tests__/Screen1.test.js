import React from 'react';
import Screen1 from '../src/Screen1';
import {describe, it, expect} from '@jest/globals';
import renderer from 'react-test-renderer';

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(() => ({})),
}));
describe('Screen1', () => {
  it('should render the same as the snapshot', () => {
    const tree = renderer.create(<Screen1 />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
