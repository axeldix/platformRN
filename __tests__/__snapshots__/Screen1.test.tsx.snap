// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen1 should render the same as the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(67,241,181,0.3)",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 50,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    Screen1
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#b8e378",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      RN Navigation -&gt; Screen 2
    </Text>
  </View>
</View>
`;
